jetbrains-core/src/software/aws/toolkits/jetbrains/core/executables/Versions.kt

```diff
@@ -0,0 +1,38 @@
+// Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package software.aws.toolkits.jetbrains.core.executables
+
+import com.intellij.util.text.SemVer
+
+interface Version : Comparable<Version> {
+    fun displayValue(): String
+}
+
+data class SemanticVersion(private val version: SemVer) : Version {
+    override fun displayValue(): String = version.toString()
+
+    override fun compareTo(other: Version): Int = version.compareTo((other as SemanticVersion).version)
```

Comment:
needs to handle other versions

Reply:
It is actually valid to throw class cast exception on compareTo. Theres no way to compare differing schemes any other way that I can think of